src/Authentication/Authentication/Cmdlets/ConnectMgGraph.cs

```diff
@@ -61,12 +62,39 @@ public class ConnectMgGraph : PSCmdlet, IModuleAssemblyInitializer, IModuleAssem
             HelpMessage = "Determines the scope of authentication context. This accepts `Process` for the current process, or `CurrentUser` for all sessions started by user.")]
         public ContextScope ContextScope { get; set; }
 
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the national cloud environment to connect to. By default, the SDK uses global cloud.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("EnvironmentName", "NationalCloud")]
+        public string Environment { get; set; }
+
         private CancellationTokenSource cancellationTokenSource;
 
+        private IGraphEnvironment environment;
+
         protected override void BeginProcessing()
         {
-            ValidateParameters();
             base.BeginProcessing();
+            ValidateParameters();
+
+            if (GraphEnvironment.BuiltInEnvironments.ContainsKey(GraphEnvironmentConstants.EnvironmentName.Global))
+            {
+                environment = GraphEnvironment.BuiltInEnvironments[GraphEnvironmentConstants.EnvironmentName.Global];
+            }
+            else
+            {
+                WriteWarning($"Default environment {GraphEnvironmentConstants.EnvironmentName.Global} cannot be found from environment list.");
```

Comment:
the actual user couldn't actually do anything about this right?  Is this even possible given this is all in memory?